flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -129,4 +130,11 @@
 	 * @return Future acknowledge if the task is successfully canceled
 	 */
 	Future<Acknowledge> cancelTask(ExecutionAttemptID executionAttemptID, @RpcTimeout Time timeout);
+
+	/**
+	 *  request heartbeat from the resource manager
```

Comment:
It does not request a heartbeat from the resource manager but it is a heartbeat request from the resource manager.